src/plugins/tonegen/dbusif.c

```diff
@@ -0,0 +1,331 @@
+/*************************************************************************
+This file is part of tone-generator
+
+Copyright (C) 2010 Nokia Corporation.
+              2014 Jolla Ltd.
+
+This library is free software; you can redistribute
+it and/or modify it under the terms of the GNU Lesser General Public
+License as published by the Free Software Foundation
+version 2.1 of the License.
+
+This library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public
+License along with this library; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301
+USA.
+*************************************************************************/
+
+#define _GNU_SOURCE
+
+#include <stdlib.h>
+#include <string.h>
+#include <stdarg.h>
+#include <errno.h>
+
+#include <ngf/request.h>
+#include <ngf/proplist.h>
+#include <ngf/log.h>
+#include <trace/trace.h>
+
+#include "tonegend.h"
+#include "dbusif.h"
+
+#define TRACE(f, args...) trace_write(trctx, trflags, trkeys, f, ##args)
+#define LOG_CAT "tonegen-dbusif: "
+
+static DBusHandlerResult handle_message(DBusConnection *,DBusMessage *,void *);
+static gchar *create_key(gchar *, gchar *, gchar *);
+static void destroy_key(gpointer);
+
+
+static char *path    = "/com/Nokia/Telephony/Tones";
```

Comment:
Can we define this in some header file and then reuse it everywhere we refer to it? This would make the "binding" more explicit, and somebody can go header-hunting for the right files and arrive at ngfd.